crates/matrix-sdk/src/authentication/qrcode/rendezvous_channel.rs

```diff
@@ -0,0 +1,433 @@
+    ///
+    /// By inbound we mean that we're going to attempt to read an initial
+    /// message from the rendezvous session on the given [`rendezvous_url`].
+    pub(super) async fn create_inbound(
+        client: HttpClient,
+        rendezvous_url: &Url,
+    ) -> Result<InboundChannelCreationResult, HttpError> {
+        // Receive the initial message, which should be empty. But we need the ETAG to
+        // fully establish the rendezvous channel.
+        let response = Self::receive_message_impl(&client.inner, None, &rendezvous_url).await?;
+
+        let etag = response.etag.clone();
+
+        let initial_message = RendezvousMessage {
+            status_code: response.status_code,
+            body: response.body,
+            content_type: response.content_type.unwrap_or_else(|| "text/plain".to_owned()),
+        };
+
+        let channel = Self { client, rendezvous_url: rendezvous_url.clone(), etag };
+
+        Ok(InboundChannelCreationResult { channel, initial_message: initial_message.body })
+    }
+
+    /// Get the URL of the rendezvous session we're using to exchange messages
+    /// through the channel.
+    pub(super) fn rendezvous_url(&self) -> &Url {
+        &self.rendezvous_url
+    }
+
+    /// Send the given `message` through the [`RendezvousChannel`] to the other
+    /// device.
+    ///
+    /// The message must be of the `text/plain` content type.
+    #[instrument(skip_all)]
+    pub(super) async fn send(&mut self, message: Vec<u8>) -> Result<(), HttpError> {
+        let etag = self.etag.clone();
+
+        let request = self
+            .client
+            .inner
+            .request(Method::PUT, self.rendezvous_url().to_owned())
+            .body(message)
+            .header(IF_MATCH, etag)
+            .header(CONTENT_TYPE, TEXT_PLAIN_CONTENT_TYPE);
+
+        debug!("Sending a request to the rendezvous channel {request:?}");
+
+        let response = request.send().await?;
+        let status = response.status();
+
+        debug!("Response for the rendezvous sending request {response:?}");
+
+        if status.is_success() {
+            // We successfully send out a message, get the ETAG and update our internal copy
+            // of the ETAG.
+            let etag = get_header(response.headers(), &ETAG)?;
+            self.etag = etag;
+
+            Ok(())
+        } else {
+            let builder = http::Response::builder().status(status);
+            let body = response.bytes().await?;
+
+            let response = builder.body(body).map_err(IntoHttpError::from)?;
+            let error = FromHttpResponseError::<RumaApiError>::Server(RumaApiError::Other(
+                MatrixError::from_http_response(response),
+            ))
+            .into();
+
+            Err(error)
+        }
+    }
+
+    /// Attempt to receive a message from the [`RendezvousChannel`] from the
+    /// other device.
+    ///
+    /// The should be of the `text/plain` content type but the parsing and
```

Comment:
```diff
- The should be of the `text/plain` content type
+ It should have the `text/plain` content type
```

?

Reply:
Fixed in https://github.com/matrix-org/matrix-rust-sdk/commit/74883c816e129fa2f643bf74c9e470a0a13491fb as well.